test/ATokenFlexVotingFork.t.sol

```diff
@@ -1510,4 +1581,478 @@ contract VoteTest is AaveAtokenForkTest {
+      _who
+    );
+    vm.stopPrank();
+
+    // Advance the clock
+    uint256 _blocksJumpedThirdTime = 10;
+    vm.roll(block.number + _blocksJumpedThirdTime);
+    vm.warp(block.timestamp + 10 days);
+
+    assertEq(
+      aToken.getPastStoredBalance(_who, block.number - _blocksJumpedThirdTime),
+      aToken.exposed_RawBalanceOf(_who)
+    );
+    assertEq(aToken.getPastStoredBalance(_who, block.number - 1), aToken.exposed_RawBalanceOf(_who));
+    assertGt(
+      _rawBalances[1], // The raw balance pre-withdrawal.
+      aToken.getPastStoredBalance(_who, block.number - _blocksJumpedThirdTime)
+    );
+  }
+
+  function test_GetPastStoredBalanceHandlesTransfers() public {
+    _initiateRebasing();
+
+    address _userA = makeAddr("GetPastStoredBalanceHandlesTransfers _userA");
+    address _userB = makeAddr("GetPastStoredBalanceHandlesTransfers _userB");
+    uint256 _amount = 4242 ether;
+
+    // Deposit.
+    _mintGovAndSupplyToAave(_userA, _amount);
+    uint256 _initRawBalanceUserA = aToken.exposed_RawBalanceOf(_userA);
+
+    // Advance the clock so that we checkpoint and let some rebasing happen.
+    vm.roll(block.number + 100);
+    vm.warp(block.timestamp + 100 days);
+
+    // Get the rebased balance for userA.
+    uint256 _initBalanceUserA = aToken.balanceOf(_userA);
+    uint256 _initBalanceUserB = aToken.balanceOf(_userB);
+    assertEq(_initBalanceUserB, 0);
+
+    // Transfer aTokens to userB.
+    vm.prank(_userA);
+    aToken.transfer(_userB, _initBalanceUserA / 3);
+    assertApproxEqAbs(aToken.balanceOf(_userA), 2 * _initBalanceUserA / 3, 1);
+    assertApproxEqAbs(aToken.balanceOf(_userB), _initBalanceUserA / 3, 1);
+
+    // Advance the clock so that we checkpoint.
+    vm.roll(block.number + 1);
+    vm.warp(block.timestamp + 1 days);
+
+    // Confirm voting weight has shifted.
+    assertApproxEqAbs(
+      aToken.getPastStoredBalance(_userA, block.number - 1),
+      2 * _initRawBalanceUserA / 3,
+      1 // In case of rounding.
+    );
+    assertApproxEqAbs(
+      aToken.getPastStoredBalance(_userB, block.number - 1),
+      _initRawBalanceUserA / 3,
+      1 // In case of rounding.
+    );
+  }
+
+  function test_GetPastStoredBalanceHandlesTransferFrom() public {
+    _initiateRebasing();
+
+    address _userA = makeAddr("GetPastStoredBalanceHandlesTransfers _userA");
+    address _userB = makeAddr("GetPastStoredBalanceHandlesTransfers _userB");
+    uint256 _amount = 4242 ether;
+
+    // Deposit.
+    _mintGovAndSupplyToAave(_userA, _amount);
+    uint256 _initRawBalanceUserA = aToken.exposed_RawBalanceOf(_userA);
+
+    // Advance the clock so that we checkpoint and let some rebasing happen.
+    vm.roll(block.number + 100);
+    vm.warp(block.timestamp + 100 days);
+
+    // Get the rebased balance for userA.
+    uint256 _initBalanceUserA = aToken.balanceOf(_userA);
+    uint256 _initBalanceUserB = aToken.balanceOf(_userB);
+    assertEq(_initBalanceUserB, 0);
```

Comment:
Comments say "userA" but we only assert for userB. Same inconsistency in previous test

Reply:
Also can we assert that user A actually has a balance? Otherwise you can't tell we're not just transferring zero everywhere

Reply:
Done in eea4149

Reply:
The comment is fixed but I don't think the assertion was changed to ensure user A has a balance?

Reply:
Oh, sorry, I think I made the change before reading your second comment.

Do we need to assert that? We know that user has an aToken balance b/c the user just supplied to aave

Reply:
Just added them anyway in 92d97c9